deltacat/compute/compactor/repartition_session.py

```diff
@@ -35,16 +35,15 @@
 logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
 
 
-# TODO:(rootliu) move this repartition function to a separate module under compute
+# TODO: move this repartition function to a separate module under compute
 def repartition(
     source_partition_locator: PartitionLocator,
     destination_partition_locator: PartitionLocator,
     repartition_args: Any,
     compaction_artifact_s3_bucket: str,
     last_stream_position_to_compact: int,
     repartition_type: RepartitionType = RepartitionType.RANGE,
-    rebase_source_partition_locator: Optional[PartitionLocator] = None,
-    rebase_source_partition_high_watermark: Optional[int] = None,
+    repartition_during_rebase: bool = True,
```

Comment:
Do we need this arg at all? I think we are adding this because of discover_deltas signature. If it's rebase source != destination, otherwise source == destination. Right?

Reply:
I think source != destination for both rebase and incremental;
rebase: source=spark, destination=ray
incremental:source=source, destination=ray

Reply:
What does incremental has to do with repartitioning? We are repartitioning all the deltas in the source and writing the output to destination. That is a simple contract.

Reply:
makes sense.

Reply:
simplified the arg and logic.